Add subscribe --dangerously-encode-payloads-as-plain-text

scothis opened the pull request:
The subscribe command base64 encodes payloads before emitting them to
the console. This can be annoying if you know the payloads are actually
plain text.

The --dangerously-encode-payloads-as-plain-text will instead treat the
payload as plain text. If the stream contains binary data, it may
corrupt your terminal.

scothis committed:
Add subscribe --dangerously-encode-payloads-as-plain-text

The subscribe command base64 encodes payloads before emitting them to
the console. This can be annoying if you know the payloads are actually
plain text.

The --dangerously-encode-payloads-as-plain-text will instead treat the
payload as plain text. If the stream contains binary data, it may
corrupt your terminal.

trisberg commented:
could we rename this to something other than `--dangerous-` and aren't we decoding the payload?

Maybe `--decode-payloads-as-plain-text` - we could add a warning to the help text to only use this when you know the payload content can be safely decoded

scothis commented:
> could we rename this to something other than --dangerous

The operation is potentially dangerous, users should know what they are getting themself into.

> aren't we decoding the payload?

It's still an encoding, instead of base64 encoding the payload bytes, we're plain text encoding

trisberg commented:
what's so dangerous about it? isn't it more of a nuisance if you use this on a binary stream? similar to using `cat` with a binary file.

sbawaska commented:
> what's so dangerous about it? isn't it more of a nuisance if you use this on a binary stream? similar to using `cat` with a binary file.

I agree, maybe restore `--payload-as-string`? This PR already adds a warning already

jldec commented:
I would propose fixing this using an --option name which is not excessively long or alarmist. I'd be fine with the original `--payload-as-string` but something different which hints toward using with care would be fine too.

scothis committed:
Rename --dangerously-encode-payloads-as-plain-text to --payload-encoding

Instead of a boolean flag, the new flag is an enum that defaults to
base64. The default behavior is unchanged

Before:
    --dangerously-encode-payloads-as-plain-text

Now:
    --payload-encoding raw

scothis commented:
Renamed `--dangerously-encode-payloads-as-plain-text` to `--payload-encoding`
    
Instead of a boolean flag, the new flag is an enum that defaults to `base64`. The default behavior is unchanged

Before:

    --dangerously-encode-payloads-as-plain-text
    
Now:

    --payload-encoding raw

trisberg reviewed (approved):
LGTM now

scothis commented:
Screwed up the message on the merge commit, oh well.
